src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -20,65 +20,52 @@ internal static class NullableAnnotationExtensions
         /// Join nullable annotations from the set of lower bounds for fixing a type parameter.
         /// This uses the covariant merging rules.
         /// </summary>
-        public static NullableAnnotation Join(this NullableAnnotation a, NullableAnnotation b)
-        {
-            if (a.IsAnnotated() || b.IsAnnotated())
-                return NullableAnnotation.Annotated;
-            return (a < b) ? a : b;
-        }
+        public static NullableAnnotation Join(this NullableAnnotation a, NullableAnnotation b) => (a < b) ? b : a;
```

Comment:
>Join [](start = 41, length = 4)

Could you update the doc comment to say something like "Annotated wins over Oblivious which wins over NotAnnotated"?
Unless people memorize the enum order, the formula does not really help understand the behavior.
(Same for `Meet` below, if you don't mind)